HW1/P5/P5_bfs.py

```diff
@@ -0,0 +1,104 @@
+from pyspark import SparkContext
+from pyspark import AccumulatorParam
+
+class SetAccumulator(AccumulatorParam):
+    # hash of node to parent
+    def zero(self, initialValue):
+        return {}
+
+    # combine the dictionaries
+    def addInPlace(self, s1, s2):
+        union = s2.copy()
+        union.update(s1)
+        return union
+
+# update parent pointers so we can backtrack
+def setParents(child, parent):
+    pointer = {}
+    for c in child:
+        pointer[c] = parent
+    return pointer
+
+def bfs(sc, graph, source, dest):
```

Comment:
I think on Piazza, we were told this needed to return all possible paths, not just one single path.